import "stage.proto";
import "common.proto";


message RankItems{
    required string nickname = 1;//角色昵称
    required int32 rank = 2;
    required int32 level = 3;
    required int32 ap = 4;
    repeated int32 hero_ids = 5;
    repeated int32 hero_levels = 6;
    required int32 head_no = 7;
    required int32 character_id = 8;
    optional int32 vip_level = 9 [default = -1];
    optional string guild_name = 10;
}

message PlayerRankResponse{
    repeated RankItems rank_items = 1;
    optional int32 player_rank = 2;
    optional int32 pvp_score = 3;
    optional int32 pvp_upstage_challenge_id = 4 [default = 0];
    optional string pvp_upstage_challenge_nickname = 5;
}

message PvpFightRequest{
    required int32 challenge_rank = 1;
    optional int32 challenge_id = 2;
    repeated int32 lineup = 3; // 队形
    optional int32 skill = 4; //无双技能ID
    optional int32 self_rank = 5;
}

message PvpFightRevenge{
    required int32 black_id = 1;
    repeated int32 lineup = 2; // 队形
    optional int32 skill = 3; //无双技能ID
}

message PvpFightOvercome{
    required int32 index = 1;
    repeated int32 lineup = 3; // 队形
    optional int32 skill = 4; //无双技能ID
}

//864
message PvpRobTreasureRequest{
    required int32 uid = 1;
    required int32 chip_id = 2;
}

message PvpFightOvercomeInfo{
    repeated int32 character_ids = 1;
    required int32 index = 2;
}

message PvpPlayerInfoRequest{
    required int32 player_rank = 1;
}

message ResetPvpTime{
	required int32 times = 1;
}

message ResetPvpOvercomeTime{
	required int32 times = 1;
}

//865
message RobTreasureMoreTimesRequest{
    required int32 uid = 1;
    required int32 chip_id = 2;
    required int32 times = 3;
}

// 进入关卡战斗返回 command:1503
message PvpFightResponse{
    optional CommonResponse res = 1;
    repeated BattleUnit red = 2;         // 红方数据 自己
    repeated BattleUnit blue = 3;        // 对方数据
    optional int32 red_skill = 4;        // 我方无双
    optional int32 red_skill_level = 5;  // 我方无双
    optional int32 blue_skill = 6;       // 敌方无双
    optional int32 blue_skill_level = 7; // 敌方无双
    optional bool fight_result = 8;      // 战斗结果
    optional int32 seed1= 9;             // 随机种子1
    optional int32 seed2= 10;            // 随机种子2
    optional GameResourcesResponse gain = 11;
    optional int32 top_rank = 12;
    optional GameResourcesResponse award = 13;
    optional GameResourcesResponse award2 = 14;
    optional int32 rank_incr = 15;
    optional GameResourcesResponse consume = 16;
    optional int32 before_rank = 17;
}


// 连续五次夺宝  command:1504
message RobTreasureMoreTimesResponse{
    required CommonResponse res = 1;
    repeated RobTreasureReward one_time_info=2;
}

// 夺宝战利品
message RobTreasureReward{
    optional GameResourcesResponse gain = 1; // 宝箱随机掉落
    repeated  GameResourcesResponse look_gain = 2;
    optional PvpFightResponse fight_info=3;
}

message PvpOvercomeAwardRequest{
	required int32 index = 1;
}

message PvpOvercomeAwardResponse{
    required CommonResponse res = 1;
    optional GameResourcesResponse gain = 2;
}

message BattleBuff{
	required int32 index = 1;
	required int32 buff_type = 2;
	required int32 value_type = 3;
	required float value = 4;
	optional int32 star = 5;
}

message GetPvpOvercomeBuffRequest{
	required int32 index = 1;
}

message GetPvpOvercomeBuffResponse{
    required CommonResponse res = 1;
	repeated BattleBuff buff = 2;
}

message BuyPvpOvercomeBuffRequest{
	required int32 index = 1;
	required int32 num = 2;
}

message BuyPvpOvercomeBuffResponse{
    required CommonResponse res = 1;
	repeated BattleBuff buff = 2;
}

message GetPvpOvercomeInfo{
    required int32 pvp_overcome_index = 1;
    required int32 stars = 2;
    required int32 refresh_count = 3;
    repeated int32 awarded = 4;
    repeated BattleBuff buff = 5;
    repeated int32 target_fight_powers = 6;
    repeated int32 target_ids = 7;
	optional bool is_failed = 8;
}
